### PR TITLE
Find student bug

### PR DESCRIPTION
**Pull request by aaron-jx-li**
# Find student bug fix

## Purpose

The purpose of this Pull Request is to fix a bug regarding the find student's page

### CODE REVIEW

**Comment by nociza** [comment]
I am able to reproduce the bug locally, it seems like some of the cucumber tests are looking explicitly for text cues that are not present after this change. It could be that one or more user routes are altered, we need to dig deeper into this.

**Comment by nociza** [comment]
LGTM

File: app/views/tutors/find_students.html.erb [hunk 1]
```diff
@@ -63,29 +63,32 @@
...
-                                                        <p> Subject: <b> <%= req.subject%> </b></p>
-                                                        <p class="last"> <button class="btn btn-primary" onclick="showEmailForm(['<%=Tutee.find(req.tutee_id).email%>', '<%=req.tutee_id%>', '<%=req.id %>', '<%=req.meeting_length %>' ])">Select Student</button></p>
-                                                    </li>
-                                                </ul>
+                                    <% if (x.nil?) %>
+                                        <div class="alert alert-info">
+                                            <p> There are no students in the queue for <%=true_class%>. </p>
+                                        </div>
+                                    <% else %>
+                                        <% course_obj = Course.where(name: true_class, semester: Course.current_semester).first %>
```
Comment (abhinavDhulipala) [inline app/views/tutors/find_students.html.erb:71]: Redundant DB query. Could harm performance. You could either alias x or just reuse x

File: app/views/tutors/find_students.html.erb [hunk 2]
```diff
@@ -63,29 +63,32 @@
                                     role="tabpanel"
                                     aria-labelledby="<%=true_class %>-tab">
                                     <% x = Course.where(name: true_class, semester: Course.current_semester).first %>
```
Comment (abhinavDhulipala) [inline app/views/tutors/find_students.html.erb:65]: Perhaps you could name this variable something more descriptive like ```coruse_obj``` instead. Or put a comment hinting at its meaning

**Review by abhinavDhulipala — COMMENTED** [review]
Code should have a consistent style and format.
Names, the indentation should follow the conventions of the language and of the code around it.
Is the code relatively easy to understand? Do method names make sense?
Don’t read just the code, look at the design!
 - Are the design patterns used appropriate?
   -  N/A
 - Is the data modeling sound?
    - N/A
 
- Does the feature make sense? Does the code address the requirements?
    - More descriptive bug description would probably help address this question. Looks like it addressed a NullException bug, but there are no cucumber/rspec to verify this. 
- Use the description along with any linked stories to give you context. It’s OK if you don’t have all the context.Temp files, local configs, scratch notes, etc.
   -  Git branch not named/linked to a pivotal tracker. We are assuming a ref to this ticket ```#177421889```
 - Are there any files that should be removed from the PR?
    - app/controllers/requests_controller.rb should probably be removed as the comments don't really add anything. 
    - app/views/requests/new.html.haml should be removed as it just contains spacing and pollutes the PR
Are the code comments logical and helpful?
- Comments should describe why. Rarely should a comment describe what is happening? Comments that describe the “what” usually mean you could choose better variable and method names! Do the comments around changed lines of code still make sense? It’s very easy for comments to become out of date.
    - No comments. N/A
- Is the code in the right places? The PR should still follow MVC
    - Technically, no. There is logic/queries in the view. Probably push the logic to a model function. and call controller level instance variables only in the view.
 - Are there any potentially dangerous actions?
     - We couldn't find any. We checked parameter injection, input querying, etc. The code might be vulnerable in the future because of all the logic being handled on the view.
 - New migrations?
     - N/A
 - Routes that can delete a lot of data?
    - N/A
 - Actions that change permissions?
     - N/A this looks like it just fetches data.
 - Are there safeguards in place? Are these any tests?
    - No, there are not really safeguards. There aren't any tests for this either.
Review Template
Paste this block as the template for your main code review. It’s written in Markdown, so it should be nicer to read when viewed on GitHub. 

**What is the goal of this PR?**
_Fix a null pointer bug in view. In the event that there are no students to fetch_

**Were you able to view screenshots, story cards, and or test the app?**
_We were able view the pivotal tracker stories_

**Is the PR in a shape that makes it easy to review?**
_It lacks a description of the bug, so we had to infer that. It also lacks a linked Pivotal tracker so we had to infer that as well_ 

**What are the status of the tests?**
Are the tests included all passing? Are the tests comprehensive? Are there any features that need to be tested? Are there any tests that are redundant?
_There are none. Coverage is probably sub-par and should be fixed_

**What is the style of the code like?**
Are there any code linters that were run automatically?
What is their status?
If there are no linters, you get to play the human code linter.
No

**Is there anything missing from this PR that should be included?**
_PV number, bug description._

**Feature Design: Are there any parts of this feature that are cause for concern?**
There might be nothing. Could this feature be abused by users? Does the code look safe in how it treats data? Are there any potential bugs that should be guarded against?
_answered in the above block._

---

### Overall Assessment

**What does this PR do well?**
_Hot fix for the bug we think..._

**What is something that could be improved with this PR?**
_Yes, MVC logic handling, better variable naming_

**Pull request merged**